Python/Product/Debugger/Debugger/DebugEngine/AD7Engine.cs

```diff
@@ -561,7 +562,7 @@ public sealed class AD7Engine : IDebugEngine2, IDebugEngineLaunch2, IDebugProgra
             _breakOnException.Clear();
             _defaultBreakOnExceptionMode = (int)enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_UNCAUGHT;
 
-            SetExceptionInfo(_defaultBreakOnExceptionMode, _breakOnException);
+            TaskHelpers.RunSynchronouslyOnUIThread(ct => SetExceptionInfoAsync(_defaultBreakOnExceptionMode, _breakOnException, ct));
```

Comment:
The main thing it fixes is the continuations continuing as part of the COM message pump, rather than blocking other events. The first attempt at this regularly deadlocked between COM calls attempting to reach the UI thread while we were dealing with another event.

That said, I believe we are assuming that this call is coming from a non-UI thread, so having an assertion (even temporarily) to confirm that we're not unnecessarily using the helper. But I don't think it hurts.